parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java

```diff
@@ -560,24 +560,26 @@ private <T> T max(Statistics<?> statistics, int id) {
   }
 
   /**
-   * Checks against older versions of Parquet statistics which may have a null count but undefined
-   * min and max statistics. Returns true if nonNull values exist in the row group but no further
-   * statistics are available.
-   *
-   * <p>We can't use {@code statistics.hasNonNullValue()} because it is inaccurate with older files
-   * and will return false if min and max are not set.
+   * Older versions of Parquet statistics which may have a null count but undefined min and max
+   * statistics. This is similar to the current behavior when NaN values are present.
    *
    * <p>This is specifically for 1.5.0-CDH Parquet builds and later which contain the different
    * unusual hasNonNull behavior. OSS Parquet builds are not effected because PARQUET-251 prohibits
    * the reading of these statistics from versions of Parquet earlier than 1.8.0.
    *
    * @param statistics Statistics to check
-   * @param valueCount Number of values in the row group
-   * @return true if nonNull values exist and no other stats can be used
+   * @return true if min and max statistics are null
    */
-  static boolean hasNonNullButNoMinMax(Statistics statistics, long valueCount) {
-    return statistics.getNumNulls() < valueCount
-        && (statistics.getMaxBytes() == null || statistics.getMinBytes() == null);
+  static boolean nullMinMax(Statistics statistics) {
+    return statistics.getMaxBytes() == null || statistics.getMinBytes() == null;
+  }
+
+  static boolean minMaxUndefined(Statistics statistics) {
+    return (!statistics.isEmpty() && !statistics.hasNonNullValue()) || nullMinMax(statistics);
```

Comment:
So for my understanding its more like other stats are defined but not min/max?  Its not strictly what the method name says, so maybe worth a javadoc?

Also I spent some time to explore this:
```
(!statistics.isEmpty() && !statistics.hasNonNullValue())
```
I noticed statistics.isEmpty() internally is (!statistics.hasNonNullValue() && !isNumNullsSet), so I think after some math it simplifies down then to : 
(stats.isNumNullSet && !stats.hasNonNullValue)

It could be clearer this way, although I guess its safer to use isEmpty if statistitcs.isEmpty definition ever changes.

Reply:
This is actually a direct reference to the logic in Parquet Statistics.java . I'm doing the same check they do which is "!IsEmpty && !nonNullValue" see

https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java#L454

Reply:
I added a comment to explain this, but I want to stick to parquet-mr at the moment. Ideally we would have some well-defined api for this but I don't believe we do.

Reply:
Sounds good to me, I was just doing a sanity check by exploding the logic (!statistics.hasNonNullValue() && !isNumNullsSet) || !statistics.hasNonNullValue() , and it checks out.  I think I saw your comment of toString() and was not sure if its that reliable as a source of truth.

Reply:
I agree with @szehon-ho's simplification of the logic, and I'm debating whether I prefer that or using `isEmpty`, which is really unclear.

I also think that we may be adding a case to unknown when it is actually known. Since the check becomes `isNumNullSet` and not `hasNonNullValue` because NaN values trigger `hasNonNullValue = false`, are we including cases where `hasNonNullValue` is actually false because the null count is zero? This would be good to make sure tests cover.

Otherwise I think the logic in this PR looks good.

Reply:
Sounds good. It's odd that this is not based on the number of values and number of nulls. Okay though, I guess it's just a strange API.

Reply:
It is quite strange